src/main/java/com/koczy/kurek/mizera/thesisbrowser/entity/Thesis.java

```diff
@@ -1,30 +1,66 @@
 package com.koczy.kurek.mizera.thesisbrowser.entity;
 
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.koczy.kurek.mizera.thesisbrowser.entity.Author;
+import com.sun.istack.internal.NotNull;
+
+import javax.persistence.*;
+import java.util.*;
+
+@Entity
 public class Thesis {
-    private long id;
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer thesisId;
+
+    @NotNull
     private String title;
-    private String author;
-    private String link;
+    private Integer citationNo;
 
-    public Thesis(long id, String title, String author, String link) {
-        this.id = id;
-        this.title = title;
-        this.author = author;
-        this.link = link;
+
+    private String linkToPDF;
+    private String pathToTXT;
+    private Date publicationDate;
+
+    @JsonIgnore
+    @ElementCollection
+    @CollectionTable(name = "relatedTheses", joinColumns = @JoinColumn(name = "thesisId"))
+    @Column(name = "relatedTheses")
+    private List<String> relatedTheses = new ArrayList<>();
+
+    @JsonIgnore
+    @ElementCollection
+    @CollectionTable(name = "keyWords", joinColumns = @JoinColumn(name = "thesisId"))
+    @Column(name = "keyWords")
+    private Set<String> keyWords = new HashSet<>();
+
+    @JsonIgnore
+    @ManyToMany(mappedBy = "theses")
+    private Set<Author> authors = new HashSet<>();
+
+
+    public Thesis() {
     }
 
-    public Thesis(String title, String author, String link) {
+    public Thesis(String title, Integer citationNo, Set<Author> authors) {
         this.title = title;
-        this.author = author;
-        this.link = link;
+        this.citationNo = citationNo;
+        this.authors = authors;
+    }
+    public Thesis(String title,String authorName, String link){
+        this.title=title;
+        this.linkToPDF=link;
+        //TODO sort this out
+        authors.add(new Author(authorName,authorName));
     }
 
-    public long getId() {
-        return id;
+    public Thesis(String title) {
+        this.title = title;
     }
 
-    public void setId(long id) {
-        this.id = id;
+    //TODO remove; only for demo ThesisDemoService.searchTheses
+    public String getAuthor(){
+        return Objects.requireNonNull(authors.stream().findFirst().orElse(null)).toString();
```

Comment:
Czy tutaj przypadkiem gdy wykona się orElse to nie poleci npe ?

Reply:
to i tak jest niepotrzebna metoda, bo będziemy zwracać autorów, a nie autora. Stworzyłem tylko po to aby nie zmieniać wszystkiego w pozostałych classach.

Reply:
Racja. To tylko loga dorzuć żeby było wiadomo kiedy się wywaliło

Reply:
Racja. To tylko loga dorzuć żeby było wiadomo kiedy się wywaliło